### PR TITLE
Feature GitHub actions

### PR DESCRIPTION
**Pull request by Robert108**
This PR adds one build action which is intended to be used as an additional status check for branch protection rule "Minimum number of approved reviews" check that exist today. 
It also modifies the auto version tagging workflow to include build and pushing merged PRs and releases to Docker Hub. 

Additional changes include the possibility to resolve hostnames instead of only numerical IP address for objects. This is needed for users running objects on the same host in which they now can specify "host.docker.internal" instead of the docker bridge gateway ip (which might differ between docker setups) in the VehicleCatalog.xosc

### CODE REVIEW

File: docs/Installation/installation.md [hunk 1]
```diff
@@ -2,11 +2,18 @@
...
 ## <a name="docker"></a> Using the docker image
-To run ATOS using the docker image, first install docker on your computer. Then, run the following command from the root directory:
+To run ATOS using the docker image, first install docker on your computer. Then, run the following command from the root repo directory:
 ```bash
 docker compose up
 ```
 
+You can also start ATOS without the compose file with docker run:
+```bash
+docker run --ipc=host --privileged -it -v ~/.astazero/ATOS/:/root/.astazero/ATOS/ -p 80:80 -p 8080:8080 -p 8081:8081 -p 8082:8082 -p 3000:3000 -p 3443:3443 -p 55555:55555 -p 443:443 -p 9090:9090 astazero/atos_docker_env:latest bash -c "source /root/atos_ws/install/setup.sh ; ros2 launch atos launch_basic.py"
```
Comment (victorjarlow) [inline docs/Installation/installation.md:16]:
```suggestion
docker run --network="host" --ipc=host --privileged -it -v ~/.astazero/ATOS/:/root/.astazero/ATOS/ astazero/atos_docker_env:latest bash -c "source /root/atos_ws/install/setup.sh ; ros2 launch atos launch_basic.py"
```

File: docs/Installation/installation.md [hunk 2]
```diff
@@ -2,11 +2,18 @@
 There are two ways of starting ATOS: using the docker image or building from source. The docker image is the easiest way to get started, but if you intend to make changes to ATOS, we recommend building from source.
 
 ## <a name="docker"></a> Using the docker image
-To run ATOS using the docker image, first install docker on your computer. Then, run the following command from the root directory:
+To run ATOS using the docker image, first install docker on your computer. Then, run the following command from the root repo directory:
 ```bash
 docker compose up
 ```
 
+You can also start ATOS without the compose file with docker run:
```
Comment (victorjarlow) [inline docs/Installation/installation.md]:
```suggestion
You can also start ATOS without the compose file, using the host's own network stack with docker run:
```

File: docker-compose.yml [hunk 3]
```diff
@@ -21,6 +21,4 @@ services:
       - "55555:55555"
       - "443:443"
       - "9090:9090" 
-      - "53241:53241"
-      - "53240:53240"
     command: bash -c "source /root/atos_ws/install/setup.sh ; ros2 launch atos launch_basic.py"
```
Comment (victorjarlow) [inline docker-compose.yml:24]: Is there some way of using the hosts networking stack to not have to do these portmappings?

File: README.md [hunk 4]
```diff
@@ -16,11 +16,19 @@ To build ATOS follow the guide below. More documentation can be found [here](htt
...
-To run ATOS using the docker image, first install docker on your computer. Then, run the following command from the root directory of the repository:
+To run ATOS using the docker image, first install docker on your computer. Then, run the following command from the root repo directory:
 ```bash
 docker compose up
 ```
 
+You can also start ATOS without the compose file with docker run:
+```bash
+docker run --ipc=host --privileged -it -v ~/.astazero/ATOS/:/root/.astazero/ATOS/ -p 80:80 -p 8080:8080 -p 8081:8081 -p 8082:8082 -p 3000:3000 -p 3443:3443 -p 55555:55555 -p 443:443 -p 9090:9090 astazero/atos_docker_env:latest bash -c "source /root/atos_ws/install/setup.sh ; ros2 launch atos launch_basic.py"
+```
```
Comment (victorjarlow) [inline README.md:33]:
```suggestion
docker run --network="host" --ipc=host --privileged -it -v ~/.astazero/ATOS/:/root/.astazero/ATOS/ astazero/atos_docker_env:latest bash -c "source /root/atos_ws/install/setup.sh ; ros2 launch atos launch_basic.py"
```

File: README.md [hunk 5]
```diff
@@ -16,11 +16,19 @@ To build ATOS follow the guide below. More documentation can be found [here](htt
 There are two ways to start using ATOS: using the docker image or building from source. The docker image is the easiest way to get started, but if you intend to make changes to ATOS, we recommend building from source.
 
 ## <a name="docker"></a> Using the docker image
-To run ATOS using the docker image, first install docker on your computer. Then, run the following command from the root directory of the repository:
+To run ATOS using the docker image, first install docker on your computer. Then, run the following command from the root repo directory:
 ```bash
 docker compose up
 ```
 
+You can also start ATOS without the compose file with docker run:
```
Comment (victorjarlow) [inline README.md:24]:
```suggestion
You can also start ATOS without the compose file, using the host's own network stack with docker run:
```

**Review by victorjarlow — APPROVED** [review]
Before merging, implement using hosts own network stack when running via docker compose up

File: docs/Installation/installation.md (hunk 1 above)
Comment (Robert108) [reply docs/Installation/installation.md:16]: I remember now that I tried this and it did not work with the VM of Docker Desktop. It seems like when you specify --network="host" with Docker Desktop the container will use the network stack of the Docker Desktop VM, not the host computer.

Comment (victorjarlow) [reply docs/Installation/installation.md:16]: Ok, might be a good idea to update the readme with two alternatives, depending on which version of docker is used.

**Review by victorjarlow — CHANGES_REQUESTED** [review]
Was a bit premature in approving, we have to understand what we can/cannot do on different docker versions and their underlying network stack implementation

**Review by victorjarlow — APPROVED** [review]

**Review by victorjarlow — APPROVED** [review]

**Comment by ceciliahernqvist** [comment]
Checked the Docker related stuff, looks good to me!

**Review by ceciliahernqvist — APPROVED** [review]

**Pull request merged**